File: java-console/src/main/java/sn/brasilburger/entity/Complement.java
package sn.brasilburger.entity;

import sn.brasilburger.entity.enums.TypeComplement;
import java.math.BigDecimal;
import java.time.LocalDateTime;

public class Complement {
    private Integer id;
    private String nom;
    private TypeComplement type;
    private BigDecimal prix;
    private String image;
    private Boolean archive;
    private LocalDateTime dateCreation;
    
    public Complement() {
        this.archive = false;
        this.dateCreation = LocalDateTime.now();
    }
    
    public Complement(String nom, TypeComplement type, BigDecimal prix) {
        this();
        this.nom = nom;
        this.type = type;
        this.prix = prix;
    }
    
    // Getters et Setters
    public Integer getId() {
        return id;
    }
    
    public void setId(Integer id) {
        this.id = id;
    }
    
    public String getNom() {
        return nom;
    }
    
    public void setNom(String nom) {
        this.nom = nom;
    }
    
    public TypeComplement getType() {
        return type;
    }
    
    public void setType(TypeComplement type) {
        this.type = type;
    }
    
    public BigDecimal getPrix() {
        return prix;
    }
    
    public void setPrix(BigDecimal prix) {
        this.prix = prix;
    }
    
    public String getImage() {
        return image;
    }
    
    public void setImage(String image) {
        this.image = image;
    }
    
    public Boolean getArchive() {
        return archive;
    }
    
    public void setArchive(Boolean archive) {
        this.archive = archive;
    }
    
    public LocalDateTime getDateCreation() {
        return dateCreation;
    }
    
    public void setDateCreation(LocalDateTime dateCreation) {
        this.dateCreation = dateCreation;
    }
    
    @Override
    public String toString() {
        return String.format("Complement[id=%d, nom='%s', type=%s, prix=%s FCFA]", 
            id, nom, type, prix);
    }
}

File: src/Entity/Complement.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity]
#[ORM\Table(name: 'complement')]
class Complement
{
    #[ORM\Id]
    #[ORM\Column(type: 'integer')]
    #[ORM\GeneratedValue]
    private ?int $id = null;

    #[ORM\Column(type: 'string', length: 100)]
    private ?string $nom = null;

    #[ORM\Column(type: 'decimal', precision: 10, scale: 2)]
    private ?string $prix = null;

    #[ORM\Column(type: 'string', length: 255, nullable: true)]
    private ?string $image = null;

    #[ORM\Column(type: 'boolean')]
    private bool $archive = false;

    public function getId(): ?int { return $this->id; }
    
    public function getNom(): ?string { return $this->nom; }
    public function setNom(string $nom): self { $this->nom = $nom; return $this; }
    
    public function getPrix(): ?string { return $this->prix; }
    public function setPrix(string $prix): self { $this->prix = $prix; return $this; }
    
    public function getImage(): ?string { return $this->image; }
    public function setImage(?string $image): self { $this->image = $image; return $this; }
    
    public function isArchive(): bool { return $this->archive; }
    public function setArchive(bool $archive): self { $this->archive = $archive; return $this; }
}
